src/ray/common/client_connection.cc

```diff
@@ -107,11 +107,26 @@ void ServerConnection::WriteBufferAsync(
     const std::vector<boost::asio::const_buffer> &buffer,
     const std::function<void(const ray::Status &)> &handler) {
   // Wait for the message to be written.
-  boost::asio::async_write(
-      socket_, buffer,
-      [handler](const boost::system::error_code &ec, size_t bytes_transferred) {
-        handler(boost_to_ray_status(ec));
-      });
+  if (RayConfig::instance().asio_event_loop_stats_collection_enabled()) {
+    auto &io_context =
```

Comment:
```suggestion
    const auto &io_context =
```

Reply:
Btw, I'm going to have to revert these since we call `io_context.RecordStart()`, which is a non-`const` member function (it might have to add an entry to the handler stats table).